### PR TITLE
Kdtree fix new

### PR DESCRIPTION
**Pull request by KulikovNikita**
Fixes for invalid stack operations and heap behavior

### CODE REVIEW

File: cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_impl.i [hunk 1]
```diff
@@ -69,88 +69,73 @@ class Stack
...
 private:
-    T * _data;
-    size_t _top;
-    size_t _count;
-    size_t _size;
-    size_t _sizeMinus1;
+    T * _data          = nullptr;
+    size_t _top        = -1;
+    size_t _sizeMinus1 = 0;
+    size_t _size       = 1;
```
Comment (averbukh) [inline cpp/daal/src/algorithms/k_nearest_neighbors/kdtree_knn_impl.i:138]: Why by default you want size == 1 instead of 0?

**Comment by SmirnovEgorRu** [comment]
@KulikovNikita, please, don't forget to merge this in 2020mnt branch.

**Pull request merged**